src/dbnode/storage/index.go

```diff
@@ -1371,6 +1371,20 @@ func (i *nsIndex) queryWithSpan(
 		return false, err
 	}
 
+	if exhaustive {
+		i.metrics.queryExhaustive.Inc(1)
+	} else {
+		i.metrics.queryNonExhaustive.Inc(1)
+
+		if opts.RequireExhaustive {
```

Comment:
Can we also add a metric here for when we return an error? I think we need to know every type of outcome since we'll be debugging what's going on with the cluster when it gets overwhelmed by queries frequently.

Perhaps we introduce "checkedQueryWithSpan(...)":
```go
func (i *nsIndex) checkedQueryWithSpan(...) (bool, error) {
	exhaustive, err := i.queryWithSpan(...)
	if err != nil {
		if exhaustive {
			i.metrics.queryExhaustiveInternalError.Inc(1)
		} else {
			i.metrics.queryNonExhaustiveInternalError.Inc(1)
		}
		return exhaustive, err
	}

	if exhaustive {
			i.metrics.queryExhaustiveSuccess.Inc(1)
		return exhaustive, nil		
	}

	if opts.RequireExhaustive {
		i.metrics.queryNonExhaustiveLimitError.Inc(1)
	return exhaustive, fmt.Errorf("query matched too many time series: require_exhaustive=%v, limit=%d, matched=%d",
		opts.RequireExhaustive,
		opts.Limit,
		results.Size())		
	}

	// Otherwise non-exhaustive but not required to be.
	i.metrics.queryNonExhaustiveSuccess.Inc(1)
	return exhaustive, nil
}

func (i *nsIndex) queryWithSpan(...) {
	// stays as was before this change
}
```

Then for metrics:
```go
	queryExhaustiveSuccess: scope.Tagged(map[string]string{
		"exhaustive": "true",
											 "result": "success"
	}).Counter("query"),
	queryExhaustiveInternalError: scope.Tagged(map[string]string{
		"exhaustive": "true",
											 "result": "error_internal"
	}).Counter("query"),
	queryNonExhaustiveSuccess: scope.Tagged(map[string]string{
		"exhaustive": "false",
											 "result": "success"
	}).Counter("query"),
	queryNonExhaustiveInternalError: scope.Tagged(map[string]string{
		"exhaustive": "false",
											 "result": "error_internal"
	}).Counter("query"),
	queryNonExhaustiveLimitError: scope.Tagged(map[string]string{
		"exhaustive": "false",
											 "result": "error_require_exhaustive"
	}).Counter("query"),
```

Reply:
Sounds good - yeah will expand the metrics to account for the error cases

Reply:
Think we probably can add to the `query(...)` wrapper func since there isn't anything happening in there besides opening a trace.

Pushed that w/ the new metrics.